source/server/listener_impl.cc

```diff
@@ -565,8 +575,11 @@ void ListenerImpl::buildUdpListenerFactory(Network::Socket::Type socket_type,
   udp_listener_config_->listener_worker_router_ =
       std::make_unique<Network::UdpListenerWorkerRouterImpl>(concurrency);
   if (udp_listener_config_->writer_factory_ == nullptr) {
-    udp_listener_config_->writer_factory_ = std::make_unique<Network::UdpDefaultWriterFactory>();
+    udp_listener_config_->writer_factory_ =
+        &(Config::Utility::getAndCheckFactoryByName<Network::UdpPacketWriterFactory>(
```

Comment:
getAndCheckFactoryByName() is out dated. Right now, we use getAndCheckFactory() which gets the factory by protobuf type.

Reply:
Oh really? Can you say more about how it's outdated? Is it deprecated somewhere? I didn't notice anything in source/common/config/utility.h but maybe I'm not looking in the right place?

Reply:
getAndCheckFactory() would match the proto type before failing over to matching the name string literal, so using it would work in both ways. @mattklein123 pointed out to me long time ago that using the well-known name of an extension (the kName here) is going to be deprecated, though I also didn't find it to be documented anywhere. But recent non-test extensions are all using the proto type.

Reply:
Curious! I'm not quite sure how to make getAndCheckFactory() work. In particular, I don't think I understand where the message comes from. That being said I was able to make getFactoryByType() work. Is that cool? If not, do you have an example I should cargo-cult from? (I spent a while doing code search on getAndCheckFactory but callers seemed to always be getting their config passed in)

Reply:
You already figured out! I commented on the non-config determined call sites.